eemeter/modeling/models/hourly_model.py

```diff
@@ -0,0 +1,199 @@
+            self.model_res_weekend = self.model_weekend.fit()
+        except ValueError:
+            self.model_weekend = None
+            self.model_res_weekend = None
+
+
+    def compute_variance(self, df):
+        weekday_df = df.loc[df['day_of_week'].isin(self.weekdays)]
+        weekend_df = df.loc[df['day_of_week'].isin(self.weekends)]
+
+        _, weekday_X = patsy.dmatrices(self.formula,
+                                       weekday_df,
+                                       return_type='dataframe')
+
+        cov = self.model_res_weekday.cov_params()
+
+        weekday_var = self.model_res_weekday.mse_resid + \
+                      (weekday_X * np.dot(cov, weekday_X.T).T).sum(1)
+
+        _, weekend_X = patsy.dmatrices(self.formula,
+                                       weekend_df,
+                                       return_type='dataframe')
+
+        cov = self.model_res_weekend.cov_params()
+        weekend_var = self.model_res_weekend.mse_resid + \
+                      (weekend_X * np.dot(cov, weekend_X.T).T).sum(1)
+
+        weekend_var = pd.Series(weekend_var, index=weekend_df.index)
+
+        weekday_var = pd.Series(weekday_var, index=weekday_df.index)
+
+        variance_df = pd.concat([weekday_var, weekend_var])
+        variance_df.sort_index()
+        return variance_df
+
+    def predict(self, df, summed=True):
+        """
+        Takes as input dataframe indexed with hour as frequency
+        and with column tempF with hourly temparatures.
+        Returns:
+            if Summed is True, then returns summed prediction and variance
+            as tuples
+            Else, tuple of two series : prediction and varianes.
+        """
+        test_df = self.add_time_day(df)
+        test_df = self.add_hdd(test_df)
+        test_df = self.add_cdd(test_df)
+
+        # We energy is what we predict, we don't require this column
+        # except in compute_variance function which deconstructs dataframe
+        # using patsy and use it to compute variancee. If this column
+        # is not there then we just construct dummy here so that
+        # compute_variance succeed.
+        if 'energy' not in test_df:
+          test_df = test_df.assign(energy=[0.0 for xx in test_df['tempF']])
+
+
+        weekday_df = test_df.loc[test_df['day_of_week'].isin(self.weekdays)]
+        weekday_pred = self.model_res_weekday.predict(weekday_df)
+
+        weekend_df = test_df.loc[test_df['day_of_week'].isin(self.weekends)]
+        weekend_pred = self.model_res_weekend.predict(weekend_df)
+
+
+        # A series DS
+        prediction = pd.concat([weekday_pred, weekend_pred])
+        prediction.sort_index()
+
+        # A Series DS
+        variance_series = self.compute_variance(test_df)
+        if summed:
+            prediction = np.sum(prediction)
+            variance = np.sum(variance_series)
+        return prediction, variance
+
+
+    def forecast(self, df):
```

Comment:
What is the functional difference between `predict` and `forecast`? They appear to be quite similar.